src/app/product-list-page/components/product-list-page/faceted-navigation/price-range-filter/price-range-filter.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, ViewEncapsulation, Input } from '@angular/core';
+import { filter } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-price-range-filter',
+  templateUrl: './price-range-filter.component.html',
+  styleUrls: ['./price-range-filter.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class PriceRangeFilterComponent {
+  @Input() public filter;
+  public rangeValue = 0;
+  public range = [0, 100];
+
+  valueChanged(e) {
+    this.rangeValue = e.target.value;
+
+    if (this.range[0] > this.range[1]) {
+      this.range.reverse();
+    }
+  }
+
+  validateMin(e) {
+    this.validateRange(e);
+    if (e.target.value >= this.range[1]) {
+      e.preventDefault();
+      this.range[0] = this.range[1];
+    }
+  }
+
+  validateMax(e) {
+    this.validateRange(e);
+    if (e.target.value <= this.range[0]) {
+      e.preventDefault();
+      this.range[1] = this.range[0];
+    }
+  }
+
+  validateRange(e) {
```

Comment:
use return instead of *else* here
```js
if (e.target.value > 100) { //this values should be dynamic from server/mock
      e.target.value = 100;
      return;
}
if (e.target.value < 0 || e.target.value === '') {
      e.target.value = 0;
}
```

<img width="184" alt="screen shot 2019-02-15 at 11 49 29" src="https://user-images.githubusercontent.com/9499109/52848610-d8092e80-3117-11e9-8c1d-fda1b64d2396.png">